tests/test_type_inference/test_initializer.py

```diff
@@ -0,0 +1,62 @@
+import astroid
+from typing import _ForwardRef
+from python_ta.transforms.type_inference_visitor import TypeInferer
+from python_ta.typecheck.base import TypeInfo, TypeFail
+from nose.tools import eq_
+
+
+def test_class_with_init():
+    program = """
+    class Foo:
+        a
+        
+        def __init__(self):
+            a = 5
+    
+    foo = Foo()
+    """
+    ti = TypeInferer()
+    ast_mod = astroid.parse(program)
+    ti.environment_transformer().visit(ast_mod)
+    ti.type_inference_transformer().visit(ast_mod)
+    for call_node in ast_mod.nodes_of_class(astroid.Call):
+        assert isinstance(call_node.inf_type.getValue(), _ForwardRef)
+        eq_(call_node.inf_type.getValue(), _ForwardRef('Foo'))
+
+
+def test_class_without_init():
+    program = """
+    class Foo:
+        a = 5
+
+    foo = Foo()
+    """
+    ti = TypeInferer()
+    ast_mod = astroid.parse(program)
+    ti.environment_transformer().visit(ast_mod)
+    ti.type_inference_transformer().visit(ast_mod)
+    for call_node in ast_mod.nodes_of_class(astroid.Call):
+        assert isinstance(call_node.inf_type.getValue(), _ForwardRef)
+        eq_(call_node.inf_type.getValue(), _ForwardRef('Foo'))
+
+
+def test_class_with_multi_init():
```

Comment:
This isn't a good test. You can't define a method with the same name twice.

Reply:
Thanks for pointing this out, in the process of trying to fix it, I discovered another bug where `__init__` functions with additional arguments aren't properly checking the number of arguments when they are called!